app/src/main/java/com/asfoundation/wallet/ui/transact/TransferNavigator.kt

```diff
@@ -0,0 +1,125 @@
+import com.asfoundation.wallet.wallet_blocked.WalletBlockedActivity
+import io.reactivex.Completable
+import java.math.BigDecimal
+
+class TransferNavigator(private val fragmentManager: FragmentManager,
+                        private val fragment: TransferFragment,
+                        private val activity: FragmentActivity,
+                        private val defaultTokenProvider: DefaultTokenProvider) {
+
+  companion object {
+    const val TRANSACTION_CONFIRMATION_REQUEST_CODE = 12344
+    const val BARCODE_READER_REQUEST_CODE = 1
+  }
+
+  fun openAppcConfirmationView(walletAddress: String, toWalletAddress: String,
+                               amount: BigDecimal): Completable {
+    return defaultTokenProvider.defaultToken.doOnSuccess {
+      with(TransactionBuilder(it)) {
+        amount(amount)
+        toAddress(toWalletAddress)
+        fromAddress(walletAddress)
+        openConfirmation(this)
+      }
+    }
+        .ignoreElement()
+  }
+
+  fun openEthConfirmationView(walletAddress: String, toWalletAddress: String,
+                              amount: BigDecimal): Completable {
+    return Completable.fromAction {
+      val transaction = TransactionBuilder(TokenInfo(null, "Ethereum", "ETH", 18))
+      transaction.amount(amount)
+      transaction.toAddress(toWalletAddress)
+      transaction.fromAddress(walletAddress)
+      openConfirmation(transaction)
+    }
+  }
+
+  fun openAppcCreditsConfirmationView(walletAddress: String,
+                                      amount: BigDecimal,
+                                      currency: TransferFragmentView.Currency): Completable {
+    return Completable.fromAction {
+      val currencyName = when (currency) {
+        TransferFragmentView.Currency.APPC_C -> fragment.getString(
+            R.string.p2p_send_currency_appc_c)
+        TransferFragmentView.Currency.APPC -> fragment.getString(R.string.p2p_send_currency_appc)
+        TransferFragmentView.Currency.ETH -> fragment.getString(R.string.p2p_send_currency_eth)
+      }
+      fragmentManager.beginTransaction()
+          .replace(R.id.fragment_container,
+              AppcoinsCreditsTransferSuccessFragment.newInstance(amount, currencyName,
+                  walletAddress))
+          .commit()
+    }
+  }
+
+  private fun openConfirmation(transactionBuilder: TransactionBuilder) {
+    val intent = Intent(fragment.context, ConfirmationActivity::class.java).apply {
+      putExtra(C.EXTRA_TRANSACTION_BUILDER, transactionBuilder)
+    }
+    fragment.startActivityForResult(intent, TRANSACTION_CONFIRMATION_REQUEST_CODE)
+  }
+
+  fun navigateBack() = activity.onBackPressed()
+
+  fun hideKeyboard() {
+    val inputMethodManager =
+        activity.getSystemService(Activity.INPUT_METHOD_SERVICE) as InputMethodManager
+    var view = activity.currentFocus
+    if (view == null) view = View(activity)
+    inputMethodManager.hideSoftInputFromWindow(view.windowToken, 0)
+  }
```

Comment:
Why is hideKeyboard() here? I know it was on a navigator before too, but it seems more view-like to me.
In other places where this function exists, it's usually on a fragment.
Is it because this is called right before a new fragment is added (in showLoading method)?